### PR TITLE
Remove ZMXServer

### PR DESCRIPTION
**Pull request by KadekM**
https://github.com/zio/zio-zmx/issues/113

### CODE REVIEW

File: src/main/scala/zio/zmx/diagnostics/ZMXServer.scala [hunk 1]
```diff
@@ -110,7 +106,7 @@ private[zmx] object ZMXServer {
       _ <- client.close
     } yield message
 
-  def make(config: ZMXConfig): ZIO[Clock with Console with ZMXParser with FiberDumpProvider, Exception, ZMXServer] = {
+  def make(config: ZMXConfig): ZManaged[Clock with Console with ZMXParser with FiberDumpProvider, Exception, Unit] = {
```
Comment (KadekM) [inline src/main/scala/zio/zmx/diagnostics/ZMXServer.scala:109]: not sure about the Unit @adamgfraser

File: src/main/scala/zio/zmx/diagnostics/ZMXServer.scala [hunk 2]
```diff
@@ -160,15 +156,15 @@ private[zmx] object ZMXServer {
       } yield ()
     }
 
-    for {
+    val channel = for {
       addr     <- addressIo
       selector <- Selector.make
       channel  <- server(addr, selector)
       _        <- putStrLn("ZIO-ZMX Diagnostics server started...")
       _        <- serverLoop(selector, channel).forever.forkDaemon
```
Comment (adamgfraser) [inline src/main/scala/zio/zmx/diagnostics/ZMXServer.scala:164]: Will this fiber be terminated when the channel is closed?

File: src/main/scala/zio/zmx/diagnostics/ZMXServer.scala (hunk 1 above)
Comment (adamgfraser) [reply src/main/scala/zio/zmx/diagnostics/ZMXServer.scala:109]: I think `Unit` makes sense here. The `acquire` action of this `ZManaged` just performs the effect of starting this concurrent process and then the `release` action shuts it down so using `ZManaged` here is valuable to control the scope even thought doesn't actually return anything.

**Review by adamgfraser — COMMENTED** [review]
Looks great! One question about making sure everything is shut down when this is done and then good to go.

File: src/main/scala/zio/zmx/diagnostics/ZMXServer.scala (hunk 2 above)
Comment (KadekM) [reply src/main/scala/zio/zmx/diagnostics/ZMXServer.scala]: I've added termination logic (that closes also selector, why is it not Managed as well?), as well as some tests that verify we can successfuly reopen on same port (thus resource dealocation was complete)

**Comment by KadekM** [comment]
@adamgfraser answered and added some tests, I'll resolve conflicts after https://github.com/zio/zio-zmx/pull/130 is in, as I assume that will generate further conflicts

**Comment by adamgfraser** [comment]
@KadekM Just merge #130 so you should be good to merge on top of that. It looks like we have one failing test though.

**Comment by KadekM** [comment]
@adamgfraser merged, tests passed now, so it seems they're somewhat flaky.

What do you think of this?
https://github.com/zio/zio-zmx/pull/128/files#diff-6877c487120e3186dfa815973f032d31e98536554aa3569afd668e284b111cf5R156

Basically, without it the tests that I had added were failing

**Comment by adamgfraser** [comment]
@KadekM I think that makes sense. You can use the `nonFlaky` test aspect to verify that the test is constantly passing now.

**Comment by KadekM** [comment]
Locally it does, I'm pushing it to see what CI has to say about it

**Comment by KadekM** [comment]
Oh well, interesting :-)

**Comment by KadekM** [comment]
@adamgfraser it seems the problem was interaction ZMXClientSpec and ZMXServerSpec, as both were opening same port in parallel

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**